### PR TITLE
#170947607 Users should be able to view and edit their profiles

### PR DESCRIPTION
**Pull request by Niyonsengaeric**
#### What does this PR do?

- Users should be able to view and edit their profiles.

#### Description of Task to be completed?

- allow the user to view his profile.
- allow a user to edit his profile.
- test the endpoint.

#### How should this be manually tested?
- run `git clone https://github.com/Stackup-Rwanda/knights-bn-frontend.git` to clone the repo.
- run `cd knights-bn-backend` in your terminal to enter into the repository.
- run `git fetch ` command to download commits, files, and refs from a remote repository into your local repo.
- run `git ft-view-and-edit-profile-170947607` to switch to this branch.
- run `npm install` to install dependencies.
- run `npm run dev-start` to start the application.
  you will see the login page then login with your email or password or log in using your social account `Google & Facebook` 
- use `http://localhost:8080/profile` to access your profile where you can edit anything you want including your Profile picture.

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

[#170947607](https://www.pivotaltracker.com/story/show/170947607)

#### Screenshots (if appropriate)

        Login page. 

Screen Shot 2020-05-12 at 17 41 21

      View Profile Image
Screen Shot 2020-05-12 at 17 16 42

     Update Profile Image by choosing a different image location.

Screen Shot 2020-05-12 at 17 45 23

     Resize the chosen Profile image then crop

Screen Shot 2020-05-12 at 17 45 40

    Updated Profile image 

Screen Shot 2020-05-12 at 17 44 45

- You can also change different displayed information.
- Click on update to confirm your change 
- You will see a message that says `Profile updated successfully`
- You cancel your changes also by clicking on cancel 

#### Questions:
N/A

### CODE REVIEW

File: src/components/combo.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+//   <div className="form-group col-md-6">
+//     <select id={id} className={className}>
+//       <option selected>{selected}</option>
+//       <option>{option}</option>
+//       <option>{option2}</option>
+//     </select>
+//   </div>
+// );
+
+// export default ComboBox;
```
Comment (Kagorora) [inline src/components/combo.js]: Remove commented file

File: __tests__/redux/reducer/userReducer.test.js [hunk 2]
```diff
@@ -19,6 +23,64 @@ describe('Test userReducer test', () => {
     expect(userReducer(state, action).data.token).toBe('abc');
   });
   it('reset password', () => {
-    expect(userReducer({}, resetUserPassword('successfully'))).toEqual({ message: 'successfully' });
+    expect(userReducer({}, resetUserPassword('successfully'))).toEqual({
+      message: 'successfully',
+    });
+  });
+  // ============================================================================
```
Comment (Kagorora) [inline __tests__/redux/reducer/userReducer.test.js]: remove comments

**Review by Kagorora — COMMENTED** [review]
Please fix the above feedback and the following:

- On my Profile Form

1. The photo looks odd and if you could put aside the change Picture Button on the right of the picture.

Screen Shot 2020-05-13 at 09 37 21


2. Gender is not being able to be updated

3. Use the calendar to choose the birthday

File: src/views/Profile.js [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+      '/edit/user/profile',
+      updateUserProfile,
+      this.state.data
+    );
+    this.setState({
+      updateSuceed: 'Profile updated successfully',
+      updatedInfos: true,
+    });
+    const result = this.props.updatedInfo;
+    console.log('_________updated info__________>>>>', result.user);
```
Comment (Kagorora) [inline src/views/Profile.js]: Remember to remove commented lines

**Review by EugeneMunya — APPROVED** [review]

**Comment by Kagorora** [comment]
@Niyonsengaeric   Check the following

1. Date(DOB) format, There is additional values that are not needed && CONSIDER USING THE CALENDAR for a better expirience.

Screen Shot 2020-05-13 at 20 27 38


2. There is a duplication while choosing gender
Screen Shot 2020-05-13 at 20 22 32

**Review by Kagorora — APPROVED** [review]

**Review by kagabof — CHANGES_REQUESTED** [review]
This file is too large src/views/Profile.js

**Review by kagabof — APPROVED** [review]

**Pull request merged**